core/variant/array.cpp

```diff
@@ -193,8 +194,12 @@ uint32_t Array::recursive_hash(int recursion_count) const {
 
 bool Array::_assign(const Array &p_array) {
 	bool can_convert = p_array._p->typed.type == Variant::NIL;
-	can_convert |= _p->typed.type == Variant::STRING && p_array._p->typed.type == Variant::STRING_NAME;
-	can_convert |= _p->typed.type == Variant::STRING_NAME && p_array._p->typed.type == Variant::STRING;
+	can_convert |= Variant::can_convert_strict(p_array._p->typed.type, _p->typed.type);
+
+	bool narrowing_conversion = _p->typed.type == Variant::INT && p_array._p->typed.type == Variant::FLOAT;
+	if (narrowing_conversion) {
+		WARN_PRINT("Narrowing conversion in array assignment (float is converted to int and loses precision).");
```

Comment:
This warning could also be added to the analyzer, since it's checking if the types are compatible anyways when you use `Array[T](base)`

Reply:
Actually, I don't think it's necessary to warn here. We don't really have this sort of warnings for runtime. If anything, it should be debug only.